src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +40,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+
+        String[] splitAddress = trimmedAddress.split(ADDRESS_SPLIT_REGEX);
+        this.block = new Block(splitAddress[BLOCK_INDEX].trim());
```

Comment:
it doesn't quite make sense to trim a `trimmedAddress`. perhaps can change the variable name to clear the confusion